picard/util/icontheme.py

```diff
@@ -30,8 +30,8 @@
         '/usr/share/pixmaps',
     ]
 
-if 'GNOME_DESKTOP_SESSION_ID' in os.environ:
-    _current_theme = os.popen('gconftool-2 -g /desktop/gnome/interface/icon_theme').read().strip() or None
+if 'DESKTOP_SESSION' in os.environ and os.environ['DESKTOP_SESSION'].lower() == 'gnome':
+    _current_theme = os.popen('gconftool-2 -g /desktop/gnome/interface/icon_theme').read().strip() or os.popen('gsettings get org.gnome.desktop.interface icon-theme').read().strip() or None
```

Comment:
Mineo, the thing is although the above has been patched, it has been deprecated. 

So it still exists as a legacy variable so that programmes that use it are not affected.
But I think we should switch to something more consistent.
 
Also the above variable exists even though I am using unity as a DE instead of gnome.